src/schema/schema_create.c

```diff
@@ -961,6 +961,42 @@ __create_data_source(
     return (dsrc->create(dsrc, &session->iface, uri, (WT_CONFIG_ARG *)cfg));
 }
 
+/*
+ * __schema_create_config_check --
+ *     Detects any invalid config combinations for schema create.
+ */
+static int
+__schema_create_config_check(
+  WT_SESSION_IMPL *session, const char *uri, const char *config, bool import)
+{
+    WT_CONFIG_ITEM cval;
+    bool is_tiered, tiered_name;
+
+    if (import && !WT_PREFIX_MATCH(uri, "file:") && !WT_PREFIX_MATCH(uri, "table:"))
+        WT_RET_MSG(session, ENOTSUP,
+          "%s: import is only supported for 'file' and 'table' data sources", uri);
+
+    /*
+     * If tiered storage is configured at the connection level and the user has not configured
+     * tiered_storage.name to be none, then the object being created is a tiered object.
+     */
+    tiered_name =
```

Comment:
Very minor: Could this be called something like `tiered_name_set`? With just `tiered_name` I expect it to be a string/char *.